logstash-filter-elasticsearch.gemspec

```diff
@@ -21,7 +21,6 @@ Gem::Specification.new do |s|
 
   # Gem dependencies
   s.add_runtime_dependency "logstash-core-plugin-api", ">= 1.60", "<= 2.99"
-  s.add_runtime_dependency 'elasticsearch', ">= 5.0.3", " <6.0.0"
```

Comment:
Whether we like it or not, the elasticsearch client library _is_ a dependency of this plugin, and while removing it here will often "work" by nature of a different plugin providing the dependency, it's also a setup for a pretty frustrating experience. For example, in [Logstash 5.3 (picked at random)](https://github.com/elastic/logstash/blob/v5.3.0/Gemfile.jruby-1.9.lock), the elasticsearch client is an _implicit_ dependency of Logstash by way of the included-by-default elasticsearch input plugin, but if that other plugin were uninstalled this plugin would likely crash at runtime.

What version of Logstash (or `logstash-core`) began explicitly having elasticsearch as its own dependency? Can we rely on that minimum version as a _proxy_ for ensuring we have an elasticsearch library instead of removing this dependency and relying on another plugin having it? Or can we loosen the requirement here to just be `>= 5.0.3` to improve the dependency manager's chances of finding one that matches?

Reply:
@yaauie The Elasticsearch dependency came into `logstash-core` in [this commit](https://github.com/elastic/logstash/pull/7284) and has been in place since [Logstash 5.5](https://github.com/elastic/logstash/blob/5.5/logstash-core/logstash-core.gemspec)

Just as a point of reference, the elasticsearch _output_ has a dependency on an unrestricted Elasticsearch gem solely as a _development_ dependency for [4 years](https://github.com/logstash-plugins/logstash-output-elasticsearch/blame/master/logstash-output-elasticsearch.gemspec#L32).

Reply:
@yaauie Regardless of the above point, I've made the change to _relax_ rather than _remove_ the dependency.